### ISSUE TITLE
BatchedDCRNN Design

### ISSUE THREAD

**Issue by ThisIsForReview** [benedekrozemberczki/pytorch_geometric_temporal#298]
Dear The Developer

I am not sure whether this is a design issue for batchedDCRNN.     It seems to me the current code does not actually conduct the operations on the batched series.  Before calling to DCRNN (putting it simple, GCN), there is a need to make edge_index + k*num_nodes, ie., make batch copies of the graphs, then conduct GCNs.  However the current code only conduct the graph operation on the first time series in the batch, others are simply applied a linear operation.

J. G.

**Comment by OckermanSethGVSU** [comment]
Thank you for pointing this out. I fixed the bug as well as introduced some optimizations aimed at improving efficiency with large batch sizes and adj matrices. I will push my changes shortly after finishing testing.

**Comment by ThisIsForReview** [comment]
Thank you, Ockerman for fixing this.

J. G.

**Comment by ThisIsForReview** [comment]
Also another suggestion.  For all the time series dataloaders, the test dataloader should be in shuffle=False, but it seems it is set to the shuffle value used for train and validation dataloader.

J. G.
